receiver/gitproviderreceiver/internal/scraper/githubscraper/helpers.go

```diff
@@ -0,0 +1,107 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package githubscraper // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/gitproviderreceiver/internal/scraper/githubscraper"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Khan/genqlient/graphql"
+	"go.uber.org/zap"
+)
+
+func (ghs *githubScraper) getRepos(
+	ctx context.Context,
+	client graphql.Client,
+	searchQuery string,
+) ([]SearchNode, int, error) {
+
+	// here we use a pointer to a string so that graphql will receive null if the
+	// value is not set since the after: $repoCursor is optional to graphql
+	var cursor *string
+	var repos []SearchNode
+	var count int
+
+	for next := true; next; {
+		r, err := getRepoDataBySearch(ctx, client, searchQuery, cursor)
+		if err != nil {
+			ghs.logger.Sugar().Errorf("error getting repo data", zap.Error(err))
+			return nil, 0, err
+		}
+		repos = append(repos, r.Search.Nodes...)
+		count = r.Search.RepositoryCount
+		cursor = &r.Search.PageInfo.EndCursor
+		next = r.Search.PageInfo.HasNextPage
+	}
+
+	return repos, count, nil
+}
+
+func (ghs *githubScraper) getBranches(
+	ctx context.Context,
+	client graphql.Client,
+	repoName string,
+	owner string,
+	defaultBranch string,
+) (int, error) {
+
+	var branchCursor *string
+
+	r, err := getBranchData(ctx, client, repoName, owner, 50, defaultBranch, branchCursor)
+	if err != nil {
+		ghs.logger.Sugar().Errorf("Error getting branch data", "error", err)
+		return 0, err
+	}
+
+	return r.Repository.Refs.TotalCount, nil
+}
+
+func add[T ~int | ~float64](a, b T) T {
```

Comment:
I can't see where this is used.

Reply:
ah yup, when i ported this over i didn't remove these functions. I can remove them and the accompanying test until we bring over the commit and pull request time metrics over.